src/Mvc/Mvc.ViewFeatures/src/RazorComponents/RazorComponentResult.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+/// <summary>
+/// An <see cref="IResult"/> that renders a Razor Component.
+/// </summary>
+public class RazorComponentResult : IResult
+{
+    /// <summary>
+    /// Constructs an instance of <see cref="RazorComponentResult"/>.
+    /// </summary>
+    /// <param name="componentType">The type of the component to render. This must implement <see cref="IComponent"/>.</param>
+    public RazorComponentResult(Type componentType)
+    {
+        // Note that the Blazor renderer will validate that componentType implements IComponent and throws a suitable
+        // exception if not, so we don't need to duplicate that logic here.
+
+        ArgumentNullException.ThrowIfNull(componentType);
+        ComponentType = componentType;
+    }
+
+    /// <summary>
+    /// Gets the component type.
+    /// </summary>
+    public Type ComponentType { get; }
+
+    /// <summary>
+    /// Gets or sets the Content-Type header for the response.
+    /// </summary>
+    public string ContentType { get; set; }
+
+    /// <summary>
+    /// Gets or sets the HTTP status code.
+    /// </summary>
+    public int? StatusCode { get; set; }
+
+    /// <summary>
+    /// Gets or sets the parameters for the component.
+    /// </summary>
+    public object Parameters { get; set; }
```

Comment:
Is this meant to be an anonymously initialized object? (`new { A = 1, B = 'c' }`)

Reply:
It's like what we support with `Html.RenderComponentAsync` - it can be an anonymously-typed object, or an `IDictionary<string, object>`. We could add explicit support for it being a `ParameterView` but I don't think many people would want to be constructing one of those.

Reply:
I'm probably not following the question, but I expected it to be the same as with any other controller action. You would have to declare the action return type as `RazorComponentResult` and then your unit test could inspect its properties. Or you could downcast to `RazorComponentResult` in the unit test. If you mean something else please clarify 😄

Reply:
Ok, I guess that I am trying to get at is that we should probably make this a `ParameterView` instance and do the conversion/cast from object to it on the constructor. Otherwise, it's really hard to do anything with the parameters other than treat them as an opaque blob

Reply:
OK, I see. TBH a `ParameterView` isn't much easier to write assertions about - it doesn't give you random access unless you convert it to a `Dictionary<string, object>`

We could expose the result of `PropertyHelper.ObjectToDictionary(yourValue)` since we're going to perform that conversion later in order to build a `ParameterView`.

Reply:
We can make it a `Dictionary<string, object>`, that will work, even if we then convert it to a ParameterView later on

Reply:
Yep, that's what I mean with "expose the result of PropertyHelper.ObjectToDictionary(yourValue)".

Following this suggestion I've changed it and added unit tests showing how you can read the parameter values from the dictionary.